app/assets/javascripts/components/plots/forms/plot_form.jsx

```diff
@@ -0,0 +1,348 @@
+import React, { Component } from 'react';
+import Plot from '../plot';
+import ButtonBar from '../../button_bar';
+import Select from '../../inputs/select';
+import InputField from '../../inputs/input_field';
+import Matrix from '../../../models/matrix';
+import Vector from '../../../models/vector';
+import { withIntegerFilter } from '../../inputs/numeric_input';
+
+/*
+This is the component to create a plot form for plot types.
+ */
+
+export default (
```

Comment:
While this construction is interesting, I might favor component-based composition rather than this style (e.g. HeatmapForm = (props) => (<PlotForm .../>) - this allows you to handle the same behavior but also add special heatmap-form specific things if you like, and is a bit easier to understand, at least for me.

Reply:
The parameters don't need to be passed another component. I think it is more explicit in constructing a PlotForm.